ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,529 @@
+            "source": [
+                "# Report Scopes"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(InitReportStorageScopeAsync))]",
+                "\npublic interface ReportScope : IMutableScope<string>, ReportingNodeFormsEntity, MonthlyPeriodFormsEntity, ScenarioFormsEntity, CurrencyFormsEntity, BasicSliceAndDiceFormsEntity, BasicFilterFormsEntity {",
+                "\n    protected Systemorph.Vertex.Pivot.Builder.Interfaces.IPivotFactory report => GetStorage().Report;",
+                "\n    protected int headerColumnWidthValue => 250;",
+                "\n",
+                "\n    HashSet<(ReportIdentity, CurrencyType)> GetDataIdentities() => GetStorage().GetIdentities((Year, Month), ReportingNode, Scenario, CurrencyType); // TODO, add filter for identities, if the property is exposed at this level",
+                "\n       ",
+                "\n    async Task<GridOptions> GetReportTaskAsync(IDataCube<ReportVariable> data) {",
+                "\n        return await report.ForDataCube(data)",
+                "\n            .WithQuerySource(workspace)",
+                "\n            .SliceRowsBy(rowSlices)",
+                "\n            .SliceColumnsBy(columnSlices)",
+                "\n            .ReportGridOptions(headerColumnWidth: headerColumnWidthValue)",
+                "\n            .ExecuteAsync();",
+                "\n    }",
+                "\n",
+                "\n    async Task InitReportStorageScopeAsync() { // This has the Async issue, but imo it should come in the future",
+                "\n       await GetStorage().InitializeReportIndependentCacheAsync();",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface Data : IMutableScope<((int year, int month) reportingPeriod, string reportingNode, string scenario, CurrencyType currencyType, (string filterName, object filterValue)[] dataFilter)> {",
+                "\n    IDataCube<ReportVariable> InputDataCube { get; set; }",
+                "\n",
+                "\n    IDataCube<ReportVariable> DataCube { get {",
+                "\n        if(InputDataCube is null) return Enumerable.Empty<ReportVariable>().ToDataCube();",
+                "\n        var filteredDataCube = (Identity.dataFilter is null || Identity.dataFilter.Length == 0) ? InputDataCube : InputDataCube.Filter(Identity.dataFilter); ",
+                "\n        if(Identity.scenario != \"Delta\") return filteredDataCube;",
+                "\n        var bestEstimateById = filteredDataCube.Where(x => x.Scenario == null).ToDictionary(x => x.ToIdentityString());",
```

Comment:
ToIdentitySting() uses reflection for every instance. 
ToStringWith() takes as input the properties which can be retrieved by reflection once and maybe stored in the storage.

Reply:
related to #241

Reply:
done